app/javascript/components/student/editor/TestRunSummaryContent.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useCallback } from 'react'
+import { TestRunStatus } from '../Editor'
+import { TestRun, Test } from './TestRunSummary'
+import { TestsList } from './TestsList'
+
+export function TestRunSummaryContent({
+  testRun,
+  onCancel,
+}: {
+  testRun: TestRun
+  onCancel: () => void
+}) {
+  switch (testRun.status) {
+    case TestRunStatus.PASS:
+    case TestRunStatus.FAIL:
+      return <TestsList tests={testRun.tests} />
+    case TestRunStatus.ERROR:
+      return (
+        <div>
+          <p>An error occurred</p>
+          <p>We got the following error message when we ran your code:</p>
+          <p>{testRun.message}</p>
+        </div>
+      )
+    case TestRunStatus.OPS_ERROR:
+      return (
+        <div>
+          <p>An error occurred</p>
+          <p>{testRun.message}</p>
+        </div>
+      )
+    case TestRunStatus.QUEUED:
+      const handleCancel = useCallback(() => {
+        onCancel()
+      }, [onCancel])
+
+      return (
+        <div>
+          <p>We've queued your code and will run it shortly.</p>
+          <button type="button" onClick={handleCancel}>
+            Cancel
+          </button>
+        </div>
+      )
+    default:
+      return <></>
```

Comment:
Don't return an empty fragment. Return`null` if you don't want to render anything.